dbms/src/IO/WriteHelpers.h

```diff
@@ -255,7 +255,7 @@ inline void writeJSONString(const char * begin, const char * end, WriteBuffer &
 }
 
 
-template <char c>
+template <char c, char symbol = '\\'>
```

Comment:
The `symbol` name is unclear.
The fact that we escape all special characters with backslash escept `c` looks strange. Is it really consistent with MySQL?

Reply:
MySQL [sys.quote_identifier](https://dev.mysql.com/doc/refman/8.0/en/sys-quote-identifier.html) only describes the content of backquotes, when I use it for special characters When escaping, it has some truncation, for example: ````SELECT sys.quote_identifier('\b') = ` ````, the following is the result of escaping special characters in MySQL directly using `AS`, but I don't Determine if this is correct (I am not sure if the mysql client is handled as I understand it):

```
:) SELECT 1 AS `\b`;
+----+
| \b |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\f`;
+----+
| \f |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\n`;
+----+
| \n |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\r`;
+----+
| \r |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\t`;
+----+
| \t |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\0`;
+----+
| \0 |
+----+
|  1 |
+----+
1 row in set (0.00 sec)

:) SELECT 1 AS `\\`;
+----+
| \\ |
+----+
|  1 |
+----+
1 row in set (0.00 sec)
```